magento/magento2#32679: setup:upgrade --dry-run does not work as expected by other dry-run-commands in Linux-environment

4ctobias opened the pull request:
- Switched --dry-run-Parameter to VALUE_NONE in UpgradeCommand.php
- Switched --dry-run-Parameter to VALUE_NONE in InstallCommand.php

### Description (*)

setup:upgrade and setup:install does not require a value of true when using the --dry-run-parameter. This is now the same beaviour for a dry run as in other Linux-Tools (eg. rsync). As before this fix using --dry-run did not start a dry run for the command, the usage of --dry-run=true was required.

### Fixed Issues (if relevant)
1. Fixes magento/magento2#32679

### Manual testing scenarios (*)
1. Change a module version
2. Run setup:upgrade --dry-run
3. The module version should not be upgraded in the setup_module-table

4ctobias committed:
magento/magento2#32679: setup:upgrade --dry-run does not work as expected by other dry-run-commands in Linux-environment

- Switched --dry-run-Parameter to VALUE_NONE in UpgradeCommand.php
- Switched --dry-run-Parameter to VALUE_NONE in InstallCommand.php

ihor-sviziev commented:
@magento run all tests

ihor-sviziev reviewed (changes requested):
Please fix integration tests

4ctobias committed:
magento/magento2#32679: setup:upgrade --dry-run does not work as expected by other dry-run-commands in Linux-environment

- Fixed test dry-run - now also arguments without value are supported

4ctobias committed:
Merge branch '2.4-develop' into fix-32679-dry-run

ihor-sviziev commented:
@magento run all tests

4ctobias committed:
magento/magento2#32679: setup:upgrade --dry-run does not work as expected by other dry-run-commands in Linux-environment

- Fixed cli-param-handling on tests for detecting parameters that should have no value - and discarding parameters that should removed for the test as no value was specified

4ctobias committed:
Merge branch 'fix-32679-dry-run' of https://github.com/4ctobias/magento2-fork into fix-32679-dry-run

ihor-sviziev commented:
@magento run all tests

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9140](https://jira.corp.magento.com/browse/ENGCOM-9140) has been created to process this Pull Request

magento-engcom-team commented:
Hi @Den4ik, thank you for the review.
[ENGCOM-9140](https://jira.corp.magento.com/browse/ENGCOM-9140) has been created to process this Pull Request

4ctobias committed:
Merge branch '2.4-develop' into fix-32679-dry-run

4ctobias committed:
Merge branch '2.4-develop' into fix-32679-dry-run

engcom-Dash committed:
Merge branch '2.4-develop' into fix-32679-dry-run

engcom-Dash commented:
Hello @4ctobias,

Thanks for the collaboration & contribution!

✔️ QA Passed
Preconditions:

Install fresh Magento 2.4-develop
Steps to reproduce
- Create a module with schema patch.
- Run `bin/magento setup:upgrade --dry-run`
Before: ✖️
DB schema is being changed.
After: ✔️
DB schema should not be changed

Builds are failed. Hence, moving this PR to Extended Testing.

Thanks

engcom-Dash commented:
@magento run all tests

engcom-Dash committed:
33332: Fix PHPCS error related to deprecation tag

engcom-Dash commented:
@magento run all tests

engcom-Dash committed:
Merge branch '2.4-develop' into fix-32679-dry-run

engcom-Dash committed:
Merge branch '2.4-develop' into fix-32679-dry-run

engcom-Dash commented:
@magento run all tests

engcom-Dash commented:
@magento run Unit Tests, Static Tests, Integration Tests, Functional Tests CE, Functional Tests B2B

engcom-Dash commented:
@magento run Static Tests, Integration Tests, Functional Tests B2B

engcom-Dash commented:
The functional B2B test failures are different in the most recent two successive builds. Additionally, the static test and Integration Tests failure are a known issue; therefore, I am moving it to Merge in Progress.

**Functional Tests B2B**
Run 1: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/33332/3073062905aea28e955ef7361b882fdb/Functional/allure-report-b2b/index.html#categories/16b25a50b4a8b68f5548cdba82ebd5f8/33a6fc63d80a2dc6/
<img width="1791" alt="image" src="https://github.com/user-attachments/assets/5221b294-98bf-498c-9e23-1c373e6c0354">
 

Run 2: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/33332/845319c70a7c4f7dca47ffa340c9b10f/Functional/allure-report-b2b/index.html#categories/16b25a50b4a8b68f5548cdba82ebd5f8/19e97d5929846bad/
<img width="1790" alt="image" src="https://github.com/user-attachments/assets/c82ea089-4ecd-49bd-99b9-9642359e78ba">
